jupyterlab_bigquery/src/components/query_history/query_history_panel.tsx

```diff
@@ -63,63 +114,177 @@ interface Props {
+  return (
+    <Paper
+      style={{
+        display: 'flex',
+        alignItems: 'stretch',
+        marginBottom: '12px',
+      }}
+      elevation={1}
+      variant="outlined"
+    >
+      <div style={{ width: '6px', backgroundColor: '#e60000' }} />
+      <div style={{ padding: '8px', display: 'flex', alignItems: 'center' }}>
+        <Error
+          fontSize="default"
+          htmlColor="rgb(230, 0, 0)"
+          className={localStyles.icon}
+        />
+        {props.errorMsg}
+      </div>
+    </Paper>
+  );
+};
+
+const QueryDetails = (props: { job: Job }) => {
+  const { details, created, errored, query } = props.job;
   const rows = [
-    { name: 'Job ID', value: `${job.project}:${job.location}.${job.id}` },
-    { name: 'User', value: job.user },
-    { name: 'Location', value: job.location },
-    { name: 'Creation time', value: job.created },
-    { name: 'Start time', value: job.started },
-    { name: 'End time', value: job.ended },
+    {
+      name: 'Job ID',
+      value: `${details.project}:${details.location}.${details.id}`,
+    },
+    { name: 'User', value: details.user },
+    { name: 'Location', value: details.location },
+    { name: 'Creation time', value: formatDate(details.created) },
+    { name: 'Start time', value: formatDate(details.started) },
+    { name: 'End time', value: formatDate(details.ended) },
     {
       name: 'Duration',
-      value: job.duration ? `${job.duration.toFixed(2)} sec` : '0.0 sec',
+      value: details.duration
+        ? `${details.duration.toFixed(1)} sec`
+        : '0.0 sec',
     },
     {
       name: 'Bytes processed',
-      value: job.bytesProcessed
-        ? formatBytes(job.bytesProcessed, 2)
-        : job.from_cache
+      value: details.bytesProcessed
+        ? formatBytes(details.bytesProcessed, 2)
+        : details.from_cache
         ? '0 B (results cached)'
         : '0 B',
     },
-    { name: 'Job priority', value: job.priority },
-    { name: 'Destination table', value: job.destination },
-    { name: 'Use legacy SQL', value: job.useLegacySql ? 'true' : 'false' },
+    { name: 'Job priority', value: details.priority },
+    { name: 'Destination table', value: details.destination },
+    { name: 'Use legacy SQL', value: details.useLegacySql ? 'true' : 'false' },
   ];
-  return <StripedRows rows={rows} />;
+  return (
+    <div>
+      <div className={localStyles.detailsTopArea}>
+        <div>
+          {!errored && `Query completed in ${details.duration.toFixed(3)} sec`}
+          <div className={localStyles.queryTime}>{formatTime(created)}</div>
+        </div>
+        <button
+          className={localStyles.openQueryButton}
+          onClick={() => {
+            const queryId = generateQueryId();
+            WidgetManager.getInstance().launchWidget(
+              QueryEditorTabWidget,
+              'main',
+              queryId,
+              undefined,
+              [queryId, query]
+            );
+          }}
```

Comment:
Right now no flags are passed, though as we start supporting legacy SQL it will be important to pass that as an extra argument here.